### PR TITLE
adding some integrations info and fixing many broken links

### PR DESCRIPTION
**Pull request by LiraLemur**
Signed-off-by: Sara <lira.lemur@gmail.com>

### Description of the Change

Added information about integrations and how to build Iroha with Ursa! 
Bonus: fixed links broken after the restructure.

### Benefits

We could announce Ursa integration because we now have info about how to use it. 
Also - fixed links

### Possible Drawbacks 

Information about integrations if not full yet - will add more after Burrow is fully integrated

### CODE REVIEW

File: docs/source/before_start/architecture.rst [hunk 1]
```diff
@@ -87,7 +87,7 @@ Distinctive features of YAC algorithm are its scalability, performance and Crash
 To ensure consistency in the network, if there are missing blocks, they will be downloaded from another peer via `Synchronizer <#synchronizer>`__.
 Committed blocks are stored in `Ametsuchi <#ametsuchi>`__ block storage.
 
-For general definition of the consensus, please check `this link <../core_concepts/glossary.html#consensus>`_.
+For general definition of the consensus, please check `this link <glossary.html#consensus>`_.
 
 
 Synchronizer
```
Comment (lebdron) [inline docs/source/before_start/architecture.rst]: Broken link on line 97, should be `peers' <glossary.html#peer>`

File: docs/source/before_start/architecture.rst [hunk 2]
```diff
@@ -15,35 +15,35 @@ Torii
...
+Ordering Gate recieves `proposals <glossary.html#proposal>`_ (potential blocks in the chain) from Ordering Service and sends them to `Simulator <#simulator>`__ for `stateful validation <glossary.html#stateful-validation>`__.
 It also requests proposal from the Ordering Service based on the consensus round.
 
 Ordering Service
 ----------------
 
-Internal Iroha component (gRPC server) that receives messages from other `peers <../core_concepts/glossary.html#peer>`__ and combines several `transactions <../core_concepts/glossary.html#transaction>`__ that have been passed `stateless validation <../core_concepts/glossary.html#stateless-validation>`__ into a `proposal <../core_concepts/glossary.html#proposal>`__.
+Internal Iroha component (gRPC server) that receives messages from other `peers <glossary.html#peer>`__ and combines several `transactions <glossary.html#transaction>`__ that have been passed `stateless validation <glossary.html#stateless-validation>`__ into a `proposal <glossary.html#proposal>`__.
 Each node has its own ordering service.
 Proposal creation could be triggered by one of the following events:
```
Comment (lebdron) [inline docs/source/before_start/architecture.rst]: Broken link on line 54, should be `environment-specific parameters <../configure/index.html#environment-specific-parameters>`

File: docs/source/build/index.rst [hunk 3]
```diff
@@ -274,8 +274,12 @@ Main Parameters
 | BENCHMARKING |                 | OFF     | Enables or disables build of the Google Benchmarks library             |
 +--------------+                 +---------+------------------------------------------------------------------------+
 | COVERAGE     |                 | OFF     | Enables or disables lcov setting for code coverage generation          |
++--------------+                 +---------+------------------------------------------------------------------------+
+| USE_LIBURSA  |                 | OFF     | Enables usage of the HL Ursa cryptography instead of the standard one  |
 +--------------+-----------------+---------+------------------------------------------------------------------------+
 
+.. note:: If you would like to use HL Ursa cryptography for your build, please install `Rust <https://www.rust-lang.org/tools/install>`_ in addition to other dependencies. Learn more about HL Ursa integration `here <integrations/index.html#hyperledger-ursa>`_.
```
Comment (lebdron) [inline docs/source/build/index.rst]:
```suggestion
.. note:: If you would like to use HL Ursa cryptography for your build, please install `Rust <https://www.rust-lang.org/tools/install>`_ in addition to other dependencies. Learn more about HL Ursa integration `here <../integrations/index.html#hyperledger-ursa>`_.
```

File: docs/source/before_start/architecture.rst (hunk 1 above)
Comment (LiraLemur) [reply docs/source/before_start/architecture.rst]: Got it. Sublime does not highlight some links for some reason.

File: docs/source/before_start/architecture.rst (hunk 2 above)
Comment (LiraLemur) [reply docs/source/before_start/architecture.rst]: Same here. Fixed.

**Pull request merged**